File: final_build/generate-manifest.js
#!/usr/bin/env node
/**
 * Generate Projects Manifest
 * 
 * This script scans the current directory for project folders
 * and generates a projects-manifest.json file.
 * 
 * Usage: node generate-manifest.js
 */

const fs = require('fs');
const path = require('path');

function generateManifest() {
    const currentDir = __dirname;
    const projects = [];

    // Get all directories in the current folder
    const items = fs.readdirSync(currentDir, { withFileTypes: true });

    items.forEach(item => {
        // Skip hidden files, node_modules, and known non-project files
        if (item.name.startsWith('.') || 
            item.name === 'node_modules' ||
            item.name === 'showcase projects page.html' ||
            item.name === 'index.html' ||
            item.name === 'index_beta.html' ||
            item.name === 'README.md' ||
            item.name === 'projects-manifest.json' ||
            item.name === 'generate-manifest.js' ||
            item.name === 'generate-manifest.py' ||
            !item.isDirectory()) {
            return;
        }

        const folderPath = item.name;
        const indexPath = path.join(currentDir, folderPath, 'index.html');

        // Check if index.html exists in the folder
        if (fs.existsSync(indexPath)) {
            // Get file stats for creation date
            const stats = fs.statSync(indexPath);
            
            projects.push({
                name: item.name.replace(/[-_]/g, ' '),
                description: `Project in ${item.name} folder`,
                folderPath: item.name,
                pagePath: 'index.html',
                icon: '🎨',
                createdDate: stats.birthtime.toISOString()
            });

            console.log(`✓ Found project: ${item.name}`);
        }
    });

    // Sort by creation date (newest first)
    projects.sort((a, b) => new Date(b.createdDate) - new Date(a.createdDate));

    // Create manifest
    const manifest = { projects };

    // Write to file
    const manifestPath = path.join(currentDir, 'projects-manifest.json');
    fs.writeFileSync(manifestPath, JSON.stringify(manifest, null, 2));

    console.log(`\n✓ Manifest generated: projects-manifest.json`);
    console.log(`✓ Total projects found: ${projects.length}`);
    console.log(`\nYou can now open showcase projects page.html to view your projects!`);
}

try {
    generateManifest();
} catch (err) {
    console.error('Error generating manifest:', err.message);
    process.exit(1);
}

File: final_build/generate-manifest.py
#!/usr/bin/env python3
"""
Generate Projects Manifest

This script scans the current directory for project folders
and generates a projects-manifest.json file.

Usage: python generate-manifest.py
"""

import os
import json
from pathlib import Path
from datetime import datetime

def generate_manifest():
    current_dir = os.path.dirname(os.path.abspath(__file__))
    projects = []

    # Get all directories in the current folder
    for item in os.listdir(current_dir):
        item_path = os.path.join(current_dir, item)

        # Skip hidden files, known non-project files
        if (item.startswith('.') or
            item == 'node_modules' or
            item == '__pycache__' or
            item in ['showcase projects page.html', 'index.html', 'index_beta.html', 
                    'README.md', 'projects-manifest.json', 
                    'generate-manifest.js', 'generate-manifest.py'] or
            not os.path.isdir(item_path)):
            continue

        folder_path = item
        index_path = os.path.join(item_path, 'index.html')

        # Check if index.html exists in the folder
        if os.path.exists(index_path):
            # Get file creation date
            stat_info = os.stat(index_path)
            created_date = datetime.fromtimestamp(stat_info.st_ctime).isoformat() + 'Z'

            projects.append({
                'name': item.replace('-', ' ').replace('_', ' '),
                'description': f'Project in {item} folder',
                'folderPath': item,
                'pagePath': 'index.html',
                'icon': '🎨',
                'createdDate': created_date
            })

            print(f'✓ Found project: {item}')

    # Sort by creation date (newest first)
    projects.sort(key=lambda x: x['createdDate'], reverse=True)

    # Create manifest
    manifest = {'projects': projects}

    # Write to file
    manifest_path = os.path.join(current_dir, 'projects-manifest.json')
    with open(manifest_path, 'w') as f:
        json.dump(manifest, f, indent=2)

    print(f'\n✓ Manifest generated: projects-manifest.json')
    print(f'✓ Total projects found: {len(projects)}')
    print(f'\nYou can now open showcase projects page.html to view your projects!')

if __name__ == '__main__':
    try:
        generate_manifest()
    except Exception as e:
        print(f'Error generating manifest: {str(e)}')
        exit(1)
